pykern/pkcli/ci.py

```diff
@@ -8,25 +8,56 @@
 :copyright: Copyright (c) 2022 RadiaSoft LLC.  All Rights Reserved.
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
 """
-from pykern.pkdebug import pkdp, pkdlog
-from pykern import pkunit
-from pykern.pkcli import test
+from pykern import pkcli
+from pykern import pkio
 from pykern import pksetup
+from pykern import pkunit
 import re
 
-
-_FILE_TYPE = re.compile(r".py$")
-_EXCLUDE_FILES = re.compile(
+_CHECK_EOF_NEWLINE_EXCLUDE_FILES = re.compile(
+    r"/ext/|node_modules/|^run/|tests/[\w\d/]+_work/|^venv/"
+)
+_CHECK_EOF_NEWLINE_FILE_EXTS = re.compile(r"\.(html|jinja|js|json|md|py|tsx|yml)$")
+_CHECK_PRINTS_EXCLUDE_FILES = re.compile(
     f".*(?:{pkunit.DATA_DIR_SUFFIX}|{pkunit.WORK_DIR_SUFFIX})/"
     + f"|^\\w+/{pksetup.PACKAGE_DATA}/"
     + r"|pkdebug[^/]*\.py$"
     + r"|(?:^|/)\."
     + r"|^run/"
+    + r"|^venv/"
 )
+_CHECK_PRINTS_FILE_EXTS = re.compile(r".py$")
 _PRINT = re.compile(r"(?:\s|^)(?:pkdp|print)\(")
 _PRINT_OK = re.compile(r"^\s*#\s*(?:pkdp|print)\(")
 
 
+def check_eof_newline():
+    """Recursively check repo for files missing newline at end of file.
+
+    Files matching _CHECK_EOF_NEWLINE_FILE_EXTS and not matching
+    _CHECK_EOF_NEWLINE_EXCLUDE_FILES will be checked.
+    """
+
+    def _error(msg):
+        pkcli.command_error("check_eof_newline: {}", msg)
+
+    res = []
```

Comment:
How could this code be shared with check_prints?

Reply:
728c88a7bcadafc9b100216121fe8d7f8ac81c72